packages/affiliates/test/mocha/affiliates.js

```diff
@@ -21,129 +32,284 @@ const devRewardsToDistribute = "50000";
+        startTime,
+        endTime,
+        empWhitelist,
+        empCreatorAddress: "creator",
+        snapshotSteps: 1,
+        collateralTokenPrices: empWhitelist.map(() => makePrices(endTime - startTime)),
+        collateralTokenDecimals: empWhitelist.map(() => 18),
+        syntheticTokenPricesWithValueCalculation: empWhitelist.map(() => makePricesWithValue(endTime - startTime)),
+        syntheticTokenDecimals: empWhitelist.map(() => 18),
+        blocks: lodash.times(endTime - startTime, i => ({ timestamp: i, number: i })),
+        balanceHistories,
+        empDeployers,
+        totalRewards
+      };
     });
-  });
-  it("getBalanceHistory", async function() {
-    this.timeout(10000);
-    const result = await affiliates.utils.getBalanceHistory(empContracts[0], startingTimestamp, endingTimestamp);
-    assert.ok(result);
-    assert.ok(result.history.length());
-  });
-  it("getAllBalanceHistory", async function() {
-    this.timeout(10000);
-    const result = await affiliates.utils.getAllBalanceHistories(empContracts, startingTimestamp, endingTimestamp);
-    assert.equal(result.length, empContracts.length);
-    result.forEach(([address, history]) => {
-      assert.ok(address);
-      assert.ok(history);
-      assert.ok(history.history.length());
+    it("should give full rewards to a single emp with balance", function() {
+      // add some balance histories to the emp contracts. this is really what ends up adjusting the distribution.
+      // this adds a single balance history event starting at time 0 for the first emp
+      balanceHistories[0][1].handleEvent(0, {
+        name: "PositionCreated",
+        // creating a position for address "aa" with 2 collateral 1 synthetic
+        args: ["aa", "2", "1"],
+        blockTimestamp: 0
+      });
+      balanceHistories[0][1].finalize();
+      const result = affiliates.utils.calculateRewards(params);
+      assert.equal(result.empPayouts["a"], totalRewards);
+    });
+    it("should split rewards equally between equally funded emps", function() {
+      // update balance history for emp a user aa
+      balanceHistories[0][1].handleEvent(0, {
+        name: "PositionCreated",
+        // creating a position for address "aa" with 2 collateral 1 synthetic
+        args: ["aa", "2", "1"],
+        blockTimestamp: 0
+      });
+      balanceHistories[0][1].finalize();
+
+      // update balance history for emp b user bb
+      balanceHistories[1][1].handleEvent(0, {
+        name: "PositionCreated",
+        // creating a position for address "bb" with 2 collateral 1 synthetic
+        args: ["bb", "2", "1"],
+        blockTimestamp: 0
+      });
+      balanceHistories[1][1].finalize();
+
+      const result = affiliates.utils.calculateRewards(params);
+      assert.equal(result.empPayouts["a"], totalRewards / 2);
+      assert.equal(result.empPayouts["b"], totalRewards / 2);
+    });
+    it("should work with an emp which had balance and expired", function() {
```

Comment:
Do we have an open issue on whether this is the desired behavior? I know we've discussed this a little, but it seems _bad_ to reward people for expired synths. Not saying we should change immediately, but could you open an issue to discuss?

Reply:
#2265